src/content/tools/hooks.md

```diff
@@ -0,0 +1,294 @@
+
+```plaintext
+📁  example_project                   // project with hooks
+    📁  hook                          // add hook scripts here
+        📄 build.dart
+    📁  lib                           // reference your assets here
+        📄 example.dart
+    📁  src                           // add native sources here
+        📄 example_native_library.c
+        📄 example_native_library.h
+    📁  test                          // test your assets here
+        📄 example_test.dart
+```
+
+## Hooks
+
+Currently, the build hook is available, but more are
+planned. To learn more, see the following.
+
+### Build hooks {:.no_toc}
+
+With build hooks, a package can do things such as
+compile or download native assets such as C or Rust libraries.
+Afterwards, these assets can be called from the Dart code of
+a package.
+
+A package's build hook is automatically invoked by the
+Dart SDK at an appropriate time during the build process.
+Build hooks are run in parallel with Dart compilation and
+might do longer running operations such as downloading or
+calling a native compiler.
+
+Use the [`build`][] function to parse the hook
+input with [`BuildInput`] and then write the hook output
+with [`BuildOutputBuilder`]. The hook should place
+downloaded and generated assets in
+[`BuildInput.sharedOutputDirectory`][].
+
+The assets produced for your package might depend on
+[`assets`][] or [`metadata`][] produced by the build hooks
+from the packages in your direct dependencies. Therefore,
+build hooks are run in the order of dependencies.
+
+[`assets`]: {{site.pub-api}}/hooks/latest/hooks/BuildInput/assets.html
+[`build`]: {{site.pub-api}}/hooks/latest/hooks/build.html
+[`BuildInput`]: {{site.pub-api}}/hooks/latest/hooks/BuildInput-class.html
+[`BuildOutputBuilder`]: {{site.pub-api}}/hooks/latest/hooks/BuildOutputBuilder-class.html
+[`BuildInput.sharedOutputDirectory`]: {{site.pub-api}}/hooks/latest/hooks/HookInput/outputDirectoryShared.html
+[`metadata`]: {{site.pub-api}}/hooks/latest/hooks/BuildInput/metadata.html
+
+## Assets
+
+Assets are the files that are produced by a hook and then
+bundled in a Dart application. Assets can be accessed
+at runtime from the Dart code. Currently, the Dart SDK can
+use the `CodeAsset` type, but more asset types are planned.
+To learn more, see the following.
+
+### CodeAsset type {:.no_toc}
+
+A [`CodeAsset`][] represents a code asset. A code asset is
+a dynamic library compiled from a language other than
+Dart, such as C, C++, Rust, or Go. `CodeAsset` is part of
+the `code_asset` package and are called with `dart:ffi` from
+the Dart code at runtime using the `@Native` annotation.
+
+[`CodeAsset`]: {{site.pub-api}}/code_assets/latest/code_assets/CodeAsset-class.html
+
+## Use a hook {: #use-hooks-assets }
+
+To add assets to your project, use a hook. For details,
+see the following sections.
+
+### Add dependencies {: #add-dependencies-hooks-assets }
+
+To use a hook, you must first add it to your `pubspec.yaml`
+dependencies.
```

Comment:
Unless there's a specific reason developers need to add these dependencies to the file manually, do suggest they use `dart pub add hooks code_assets` instead. It's what we've been moving to elsewhere and is less prone to mistakes.